hudi-aws/src/main/java/org/apache/hudi/aws/sync/AWSGlueCatalogSyncClient.java

```diff
@@ -141,105 +156,208 @@ public AWSGlueCatalogSyncClient(HiveSyncConfig config) {
     this.databaseName = config.getStringOrDefault(META_SYNC_DATABASE_NAME);
     this.skipTableArchive = config.getBooleanOrDefault(GlueCatalogSyncClientConfig.GLUE_SKIP_TABLE_ARCHIVE);
     this.enableMetadataTable = Boolean.toString(config.getBoolean(GLUE_METADATA_FILE_LISTING)).toUpperCase();
+    this.allPartitionsReadParallelism = config.getIntOrDefault(ALL_PARTITIONS_READ_PARALLELISM);
+    this.changedPartitionsReadParallelism = config.getIntOrDefault(CHANGED_PARTITIONS_READ_PARALLELISM);
+    this.changeParallelism = config.getIntOrDefault(PARTITION_CHANGE_PARALLELISM);
   }
 
-  @Override
-  public List<Partition> getAllPartitions(String tableName) {
+  private List<Partition> getPartitionsSegment(Segment segment, String tableName) {
     try {
-      return getPartitions(GetPartitionsRequest.builder()
-              .databaseName(databaseName)
-              .tableName(tableName));
+      List<Partition> partitions = new ArrayList<>();
+      String nextToken = null;
+      do {
+        GetPartitionsResponse result = awsGlue.getPartitions(GetPartitionsRequest.builder()
+            .databaseName(databaseName)
+            .tableName(tableName)
+            .segment(segment)
+            .nextToken(nextToken)
+            .build()).get();
+        partitions.addAll(result.partitions().stream()
+            .map(p -> new Partition(p.values(), p.storageDescriptor().location()))
+            .collect(Collectors.toList()));
+        nextToken = result.nextToken();
+      } while (nextToken != null);
+      return partitions;
     } catch (Exception e) {
       throw new HoodieGlueSyncException("Failed to get all partitions for table " + tableId(databaseName, tableName), e);
     }
   }
 
-  @Override
-  public List<Partition> getPartitionsByFilter(String tableName, String filter) {
-    try {
-      if (filter.length() <= GLUE_EXPRESSION_MAX_CHARS) {
-        LOG.info("Pushdown filters: {}", filter);
-        return getPartitions(GetPartitionsRequest.builder()
-              .databaseName(databaseName)
-              .tableName(tableName)
-              .expression(filter));
-      } else {
-        LOG.warn("Falling back to listing all partition since expression filter length > {}", GLUE_EXPRESSION_MAX_CHARS);
-        return getAllPartitions(tableName);
-      }
-    } catch (Exception e) {
-      throw new HoodieGlueSyncException("Failed to get partitions for table " + tableId(databaseName, tableName) + " from expression: " + filter, e);
-    }
-  }
-
   private List<Partition> getPartitions(GetPartitionsRequest.Builder partitionRequestBuilder) throws InterruptedException, ExecutionException {
```

Comment:
This method is never used, should we remove it?

Reply:
yes, this is part of the aforementioned dilemma, but in the current view it's not needed so I deleted it.